src/mito2/src/region/opener.rs

```diff
@@ -120,6 +126,7 @@ impl RegionOpener {
         match self.maybe_open(config, wal).await {
             Ok(Some(region)) => {
                 let recovered = region.metadata();
+                // Safety: must exist.
```

Comment:
It panics if the metadata is None. So we should not add the safety note as we need the caller to ensure the metadata is set. Other safety notes in the opener are the same as this.